ntbs-service/Pages/Notifications/Edit/Items/SocialContextBase.cshtml.cs

```diff
@@ -0,0 +1,121 @@
+        [BindProperty]
+        public FormattedDate FormattedDateTo { get; set; }
+
+        public SocialContextBaseModel(
+            INotificationService service,
+            IAuthorizationService authorizationService,
+            INotificationRepository notificationRepository,
+            IItemRepository<T> socialContextRepository) : base(service, authorizationService, notificationRepository)
+        {
+            _socialContextRepository = socialContextRepository;
+        }
+
+        protected void FormatDatesForGet(T model)
+        {
+            FormattedDateFrom = model.DateFrom.ConvertToFormattedDate();
+            FormattedDateTo = model.DateTo.ConvertToFormattedDate();
+        }
+
+        protected async Task ValidateAndSave(T model, string modelName)
+        {
+            model.NotificationId = NotificationId;
+            model.Dob = Notification.PatientDetails.Dob;
+            SetDates(model, modelName);
+            model.SetFullValidation(Notification.NotificationStatus);
+
+            if (TryValidateModel(model, modelName))
+            {
+                if (RowId == null)
+                {
+                    await _socialContextRepository.AddAsync(model);
+                }
+                else
+                {
+                    model.SetModelId(RowId.Value);
+                    await _socialContextRepository.UpdateAsync(Notification, model);
+                }
+            }
+        }
+
+        public async Task<IActionResult> OnPostDeleteAsync()
+        {
+            Notification = await GetNotificationAsync(NotificationId);
+            if (!(await AuthorizationService.CanEdit(User, Notification)))
+            {
+                return ForbiddenResult();
+            }
+
+            var model = GetSocialContextBaseById(Notification, RowId.Value);
+            if (model == null)
+            {
+                return NotFound();
+            }
+
+            await _socialContextRepository.DeleteAsync(model);
+
+            return RedirectAfterSaveForNotified();
+        }
+
+        private void SetDates(T model, string modelName)
+        {
+            // The required date will be marked as missing on the model, since we are setting it manually, rather than binding it
+            ModelState.Remove($"{modelName}.DateFrom");
+            ValidationService.TrySetFormattedDate(model, modelName, nameof(model.DateFrom), FormattedDateFrom);
+            ModelState.Remove($"{modelName}.DateTo");
+            ValidationService.TrySetFormattedDate(model, modelName, nameof(model.DateTo), FormattedDateTo);
+        }
+
+        public ContentResult OnGetValidateSocialContextProperty(string key, string value, bool shouldValidateFull)
+        {
+            return ValidationService.ValidateModelProperty<SocialContextVenue>(key, value, shouldValidateFull);
+        }
+
+        public ContentResult OnGetValidateSocialContextDate(string key, string day, string month, string year)

```

Comment:
Would it be useful for `day` `month` and `year` to be `typed as `int`s instead?

Reply:
I think when I first started implementing dates I was playing around with int inputs, but then things like converting blank to "0" happen which can lead to weird behaviour. All the formatted date methods expect string inputs at the moment, so I think it might be safest to leave it for now? If we want to change it all to ints would need a bit more careful thought, probably shouldn't be part of the markups.